src/routes/userRoutes.js

```diff
@@ -0,0 +1,29 @@
+const express = require('express');
+const { getUser, signupUser, loginUser, logoffUser, getAllUsers, resetPassword, updateUserProfile } = require('../controllers/userController');
+const { verifyToken } = require('../middlewares/verifyToken');
+
+const router = express.Router();
+
+// route to get all users records
+router.get('/', verifyToken, getAllUsers);
```

Comment:
https://blog.dreamfactory.com/best-practices-for-naming-rest-api-endpoints

Continuing my point from our meeting this week, I think we should modify these routes to be more adherent to "RESTful" design. It's not super important that you become an expert tonight about this, but most importantly this means organizing our routes around "nouns" instead of "verbs".

Writing out the full route, I would lean towards:
```
GET /user/:uid                       // Get a user object by UID
POST /user                           // Create a new user object
PATCH /user/:uid                  // Update one or more fields on a user. Can be used for profile pic, email, etc.
POST /user/:uid/password   // Update the password using that google API. Would be great if we can merge      
                                             // this with PATCH /user/:uid for simplicity of the frontend team

POST /user/login                  // Attempt to log in, creating a new token if successful
DELETE /user/login              // Log off the user by deleting their token
```

Here, we center these routes around the user object and take advantage of different methods to accomplish different things. If we need to do extra things using these users, we could add additional routes like
```
/user/:uid/register/:courseid
... ETC. ...
```

Reply:
I have updated all the routes. For example, '/:uid' has been changed to '/user/:uid'